pkg/collector/corechecks/servicediscovery/usm/service.go

```diff
@@ -219,7 +222,15 @@ func ExtractServiceMetadata(args []string, envs map[string]string, fs fs.SubFS,
 
 	exe = normalizeExeName(exe)
 
-	if detectorProvider, ok := binsWithContext[exe]; ok {
+	// First try language detectors
```

Comment:
Shouldn't the specific detectors be checked first since they have more info than just the language? For example, gunicorn will be recognized as Python (the test cases in `service_test.go` should also reflect this by adding the `language: Python` there) due to the exe pointing to Python but the command line will start with gunicorn since gunicorn replaces the command line with its own string.

Reply:
Good catch, will invert the two